Macro generate ScalarTypeToCPPType, including all ScalarTypes.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38073 Move (most) generated return statements for TH functions out of the switch.
* #38072 Remove codegen for IntArrayRefStride, which isn't used.
* **#38071 Macro generate ScalarTypeToCPPType, including all ScalarTypes.**
* #37958 Kill resize-ing and zero-ing from codegen.

Fixes: https://github.com/pytorch/pytorch/issues/34826

Differential Revision: [D21476009](https://our.internmc.facebook.com/intern/diff/D21476009)

gchanan committed:
Macro generate ScalarTypeToCPPType, including all ScalarTypes.

Fixes: https://github.com/pytorch/pytorch/issues/34826

[ghstack-poisoned]

mruberry reviewed (approved):
Nice simplification!

gchanan committed:
Update on "Macro generate ScalarTypeToCPPType, including all ScalarTypes."


Fixes: https://github.com/pytorch/pytorch/issues/34826

[ghstack-poisoned]

gchanan committed:
Update on "Macro generate ScalarTypeToCPPType, including all ScalarTypes."


Fixes: https://github.com/pytorch/pytorch/issues/34826

Differential Revision: [D21476009](https://our.internmc.facebook.com/intern/diff/D21476009)

[ghstack-poisoned]
